src/stack_height/max_height.rs

```diff
@@ -173,7 +173,7 @@ pub(crate) fn compute(func_idx: u32, module: &elements::Module) -> Result<u32, E
 
 	// Add implicit frame for the function. Breaks to this frame and execution of
 	// the last end should deal with this frame.
-	let func_arity: u32 = if func_signature.return_type().is_some() {
+	let func_arity: u32 = if !func_signature.results().is_empty() {
```

Comment:
arity of the function is the number of results it returns. Before it was special casing based on if there is a result type but now `results` is an slice and we can use its length here